improver_tests/utilities/test_save.py

```diff
@@ -228,6 +228,23 @@ def test_remove_least_significant_digit(self):
         cube = load_cube(self.filepath)
         self.assertNotIn("least_significant_digit", cube.attributes)
 
+    def test_fill_value_no_mask(self):
+        """Test that fill_value is not set if there is no masked data."""
+        save_netcdf(self.cube, self.filepath)
+        cube = load_cube(self.filepath)
+        self.assertEqual(
+            cube.data.get_fill_value(), 1e20
+        )  # Default fill value for float32
+
+    def test_fill_value_with_mask(self):
+        """Test that fill_value can be overriden."""
+        cube = self.cube.copy()
```

Comment:
I think we need another test for cubes of integer data, to simulate what happens with the Categorical module.